src/System.Net.HttpListener/src/System/Net/HttpListener.cs

```diff
@@ -2,14 +2,159 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections;
 using System.Security.Authentication.ExtendedProtection;
+using System.Threading.Tasks;
 
 namespace System.Net
 {
     public sealed unsafe partial class HttpListener : IDisposable
     {
         public delegate ExtendedProtectionPolicy ExtendedProtectionSelector(HttpListenerRequest request);
 
+        private readonly object _internalLock;
```

Comment:
The filename is HttpListener.cs, all windows and Unix specific files have .windows.cs, and .unix.cs filename endings. Will put these in separate folders for organisation.